pgjdbc/src/test/java/org/postgresql/test/jdbc4/jdbc41/RangeTypeTest.java

```diff
@@ -0,0 +1,427 @@
+        // simple mode supports no bind parameter
+        stmt.executeUpdate("INSERT INTO table1 (int8range_column) VALUES ('" + int8range + "')");
+      }
+    } else {
+      insert.setObject(1, int8range);
+      insert.executeUpdate();
+    }
+
+
+    try (Statement stmt = con.createStatement();
+         ResultSet rs = stmt.executeQuery(TestUtil.selectSQL("table1", "int8range_column"))) {
+      assertTrue(rs.next());
+      assertEquals(expected, rs.getObject(1, PGint8range.class).getValue());
+      assertFalse(rs.next());
+      stmt.executeUpdate("DELETE FROM table1");
+    }
+  }
+
+  @Test
+  public void selecNumRange() throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'(,)'"));
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'[-9223372036854775808.8,9223372036854775807.7)'"));
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'(,3.3)'"));
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'[4.4,20.20)'"));
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'(4.4,20.20]'"));
+      stmt.executeUpdate(TestUtil.insertSQL("table1", "numrange_column", "'[50.50,)'"));
+
+      try (ResultSet rs = stmt.executeQuery(TestUtil.selectSQL("table1", "numrange_column", null, "numrange_column"))) {
+
+        assertTrue(rs.next());
+        PGnumrange range = rs.getObject("numrange_column", PGnumrange.class);
+        // (,)
+        assertFalse(range.isLowerInclusive());
+        assertTrue(range.isLowerInfinite());
+        assertFalse(range.isUpperInclusive());
+        assertTrue(range.isUpperInfinite());
+
+        assertTrue(rs.next());
+        range = rs.getObject("numrange_column", PGnumrange.class);
+        // [-9223372036854775808.8,9223372036854775807.7)
+        assertTrue(range.isLowerInclusive());
+        assertFalse(range.isLowerInfinite());
+        assertEquals(new BigDecimal("-9223372036854775808.8"), range.getLowerBound());
+        assertEquals(new BigDecimal("9223372036854775807.7"), range.getUpperBound());
+        assertFalse(range.isUpperInclusive());
+        assertFalse(range.isUpperInfinite());
+
+        assertTrue(rs.next());
+        range = rs.getObject("numrange_column", PGnumrange.class);
+        // (,3.3)
+        assertFalse(range.isLowerInclusive());
+        assertTrue(range.isLowerInfinite());
+        assertEquals(new BigDecimal("3.3"), range.getUpperBound());
+        assertFalse(range.isUpperInclusive());
+        assertFalse(range.isUpperInfinite());
+
+        assertTrue(rs.next());
+        range = rs.getObject("numrange_column", PGnumrange.class);
+        // [4.4,20.20)
+        assertTrue(range.isLowerInclusive());
+        assertFalse(range.isLowerInfinite());
+        assertEquals(new BigDecimal("4.4"), range.getLowerBound());
+        assertEquals(new BigDecimal("20.20"), range.getUpperBound());
```

Comment:
Could you please add `.getString` for numrange cases?
There might be `getString returning 20.20 vs 20.2 in binary vs text modes` issue.